runtime/tests/staking_reward.rs

```diff
@@ -388,70 +272,61 @@ fn staking_validators_should_receive_equal_transaction_fee_reward() {
 			let r = Executive::apply_extrinsic(xt);
 			assert!(r.is_ok());
 
-			// Check if the transfer is successful
-			assert_eq!(
-				<GenericAsset as MultiCurrency>::free_balance(&alice(), Some(CENTRAPAY_ASSET_ID)),
-				balance_amount - transfer_amount - fee
-			);
-			assert_eq!(
-				<GenericAsset as MultiCurrency>::free_balance(&bob(), Some(CENTRAPAY_ASSET_ID)),
-				balance_amount + transfer_amount
-			);
+			let issuance_after_fees_burned = RewardCurrency::total_issuance();
+			assert_eq!(issuance_after_fees_burned, initial_issuance - tx_fee);
+
+			// reward is fees * inflation
+			let total_payout = Rewards::calculate_next_reward_payout();
+			assert_eq!(total_payout, Rewards::inflation_rate().saturating_mul_acc_int(tx_fee));
+
+			// treasury should be empty until rewarded
+			assert!(RewardCurrency::free_balance(&Treasury::account_id()).is_zero());
+
+			// Submit a tx, rotate era, check rewards are paid
+			start_era(1);
 
 			// Check if stash account balances are not yet changed
+			let per_fee_reward: Balance = (Perbill::one().saturating_sub(Rewards::development_fund_take()))
+				* total_payout / validators.len() as Balance;
 			for (stash, _controller, _, _, _, _) in &validators {
-				assert_eq!(
-					<GenericAsset as MultiCurrency>::free_balance(&stash, Some(CENTRAPAY_ASSET_ID)),
-					balance_amount
-				);
+				assert_eq!(RewardCurrency::free_balance(&stash), initial_balance + per_fee_reward);
 			}
 
-			let total_issuance = GenericAsset::total_issuance(CENTRAPAY_ASSET_ID);
-			start_era(2);
-			let issued_fee_reward = per_fee_reward * validator_len; // Don't use "fee" itself directly
-			let (staking_payout, max_payout) = Staking::current_total_payout(total_issuance + issued_fee_reward);
-			let per_staking_reward = staking_payout / validator_len;
-
-			// Check total issuance of Spending Asset updated after new era
+			// treasury gets it's cut
+			let treasury_cut = Rewards::development_fund_take() * total_payout;
+			let validator_cut = per_fee_reward * validators.len() as Balance;
+			let remainder = total_payout - treasury_cut - validator_cut;
 			assert_eq!(
-				GenericAsset::total_issuance(CENTRAPAY_ASSET_ID),
-				total_issuance + max_payout + issued_fee_reward,
+				RewardCurrency::free_balance(&Treasury::account_id()),
+				treasury_cut + remainder
 			);
 
-			// Check if validator balance changed correctly
-			for (stash, _controller, _, _, _, _) in validators {
-				// Check tx fee reward went to the stash account of validator
-				assert_eq!(
-					<GenericAsset as MultiCurrency>::free_balance(&stash, Some(CENTRAPAY_ASSET_ID)),
-					balance_amount + per_fee_reward + per_staking_reward
-				);
-			}
+			// Check total issuance of spending asset updated after new era
+			assert_eq!(
+				RewardCurrency::total_issuance(),
+				issuance_after_fees_burned + total_payout
+			);
 		});
 }
 
 #[test]
-#[ignore]
 /// This tests if authorship reward of the last block in an era is factored in.
```

Comment:
changed these authorship tests to focus on the points rewarded for the block vs. the reward.